pennylane/workflow/jacobian_products.py

```diff
@@ -678,7 +678,14 @@ def compute_vjp(self, tapes: Batch, dy: Tuple[Tuple[TensorLike]]) -> Tuple:
             logger.debug("compute_vjp called with (%s, %s)", tapes, dy)
         numpy_tapes = tuple(qml.transforms.convert_to_numpy_parameters(t) for t in tapes)
         dy = qml.math.unwrap(dy)
-        return self._device.compute_vjp(numpy_tapes, dy, self._execution_config)
+        vjps = self._device.compute_vjp(numpy_tapes, dy, self._execution_config)
+        res = []
+        for t, r in zip(tapes, vjps):
+            if len(t.trainable_params) == 1 and qml.math.shape(r) == ():
+                res.append((r,))
+            else:
+                res.append(r)
+        return res
```

Comment:
Could we instead make this a constraint on `Device.compute_vjp`?  I think it it reasonable to add the constraint that the pytree structure of a VJP should match the structure of `tape.get_parameters()`.

Reply:
Yep, I'll leave it like this for now, and we can update `LightningQubit` accordingly and revert this after that change is merged.